src/modules/sensors/vehicle_angular_velocity/VehicleAngularVelocity.cpp

```diff
@@ -103,84 +104,107 @@ VehicleAngularVelocity::SensorBiasUpdate(bool force)
 bool
 VehicleAngularVelocity::SensorCorrectionsUpdate(bool force)
 {
-	if (_sensor_selection_sub.updated() || force) {
-		sensor_selection_s sensor_selection;
-
-		if (_sensor_selection_sub.copy(&sensor_selection)) {
-			if (_selected_sensor_device_id != sensor_selection.gyro_device_id) {
-				_selected_sensor_device_id = sensor_selection.gyro_device_id;
-				force = true;
-			}
-		}
-	}
-
 	// check if the selected sensor has updated
 	if (_sensor_correction_sub.updated() || force) {
 
 		sensor_correction_s corrections{};
 		_sensor_correction_sub.copy(&corrections);
 
-		// TODO: should be checking device ID
-		if (_selected_sensor == 0) {
+		// find in sensor corrections
+		if ((_sensor_correction_index < 0)
+		    || (corrections.gyro_device_id[_sensor_correction_index] != _selected_sensor_device_id)) {
+
+			for (int correction_index = 0; correction_index < MAX_SENSOR_COUNT; correction_index++) {
+				if (corrections.gyro_device_id[correction_index] == _selected_sensor_device_id) {
+					_sensor_correction_index = correction_index;
+				}
+			}
+		}
+
+		if (_sensor_correction_index == 0) {
 			_offset = Vector3f{corrections.gyro_offset_0};
 			_scale = Vector3f{corrections.gyro_scale_0};
 
-		} else if (_selected_sensor == 1) {
+		} else if (_sensor_correction_index == 1) {
 			_offset = Vector3f{corrections.gyro_offset_1};
 			_scale = Vector3f{corrections.gyro_scale_1};
 
-		} else if (_selected_sensor == 2) {
+		} else if (_sensor_correction_index == 2) {
 			_offset = Vector3f{corrections.gyro_offset_2};
 			_scale = Vector3f{corrections.gyro_scale_2};
 
 		} else {
 			_offset = Vector3f{0.0f, 0.0f, 0.0f};
 			_scale = Vector3f{1.0f, 1.0f, 1.0f};
 		}
+	}
+
+	return false;
```

Comment:
Why `false`?

Reply:
I think this can be made to return `void` .. which means `Start()` could also return `void` seeing as how it's return value is unused.